### PR TITLE
[App] Automate missing requirements installation for CLI

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds support for installing ClientCommand requirements when missing.

This is done during the connect phase. If the user provides `--yes`, it is automatically installed.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/lightning_app/cli/commands/app_commands.py [hunk 1]
```diff
@@ -1,13 +1,38 @@
...
+                requirements.add(req)
+
+    if requirements:
+        missing_requirements = []
+        for req in requirements:
+            if not package_available(req):
+                missing_requirements.append(req)
+
+        if missing_requirements:
+            print(f"Installing missing requirements: {missing_requirements}")
```
Comment (ethanwharris) [inline src/lightning_app/cli/commands/app_commands.py:30]: Should probably print the requirements that are needed and get the user to confirm that they want them installed with pip. That way they could exit and install with conda or whatever themselves if needed

**Review by ethanwharris — COMMENTED** [review]
Small suggestion :smiley:

File: src/lightning_app/cli/commands/app_commands.py (hunk 1 above)
Comment (manskx) [reply src/lightning_app/cli/commands/app_commands.py:30]: I agree, similar to `lightning install <app/component>` we show this:

```  
The installation runs the following command for you:
pip install <requirements>
Press enter to continue: y/n
```

**Review by Borda — APPROVED** [review]

**Comment by tchaton** [comment]
@Borda yes, it will.

File: src/lightning_app/cli/commands/app_commands.py (hunk 1 above)
Comment (tchaton) [reply src/lightning_app/cli/commands/app_commands.py]: Done !

**Pull request merged**